feat(docs): adds ownership and label documentation (#1223)
cloudoperators/greenhouse: docs/getting-started/operations/ownership.md

---
title: "Ownership"
weight: 3
---

> ## What is Ownership with Greenhouse

Reviewer: ```suggestion
## What is Ownership within Greenhouse
```

`Ownership` in Greenhouse is the combination of two of the core features:

- User and Organization management via [Teams](./../core-concepts/teams.md)

> - Resource deployment via [Plugins](./../core-concepts/plugins.md) to remote [Clusters](./../core-concepts/clusters.md)

Reviewer: ```suggestion
- Deployment of resources ([Plugins](./../core-concepts/plugins.md), [TeamRoleBindings](./../core-concepts/plugins.md)) to remote [Clusters](./../core-concepts/teams/#team-rbac)
```

Greenhouse provides a `1:1` relationship between a `Team` and

- `PluginPresets`
- `Plugins`
- `Clusters`
- `TeamRoleBindings`
- `Secrets`

We call this relationship `Ownership`.

## Why Ownership of Resources

[Operational processes](processes.md) facilitated via Greenhouse rely on `Ownership`:

By identifying the owner of a resource it is possible to route operational tasks on the resource to the owner.

## How is Ownership achieved

Greenhouse expects a `label` with the key `greenhouse.sap/owned-by` with a value matching an existing `Team` on the following resources in the Greenhouse central cluster:

- `PluginPresets`*
- `Plugins`*
- `Clusters`*

> - `TeamRoleBindings`*

Reviewer: Typo: additional `*`-suffix

Author: changed

- `Secrets`

and on all

- k8s resources (e.g. `Deployments`, `Pods`, ...) exposing metrics on the remote clusters

> > *Presence of the `greenhouse.sap/owned-by` label is enforced via a validating webhook. Value needs to match a valid `Team.name`.

Reviewer: ```suggestion
> *Presence of the `greenhouse.sap/owned-by` label is enforced via a validating webhook. Value needs to match a valid Team's `.metadata.name`.
```

Author: Need to change this as this is no longer the behavior

### Label Transport

> #### On Greenhouse central Cluster

Reviewer: ```suggestion
#### On Greenhouse central cluster
```

The Greenhouse controller [transports labels from a source resource to a target resource](https://github.com/cloudoperators/greenhouse/blob/main/internal/lifecycle/propagation.go) on the Greenhouse cluster.
This is currently active for:

> - `Secrets` that are used to bootsrap a `Cluster`

Reviewer: ```suggestion
- `Secrets` that are used to bootstrap a `Cluster`
```
